stan/math/opencl/prim/tail.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef STAN_MATH_OPENCL_PRIM_TAIL_HPP
+#define STAN_MATH_OPENCL_PRIM_TAIL_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/matrix_cl.hpp>
+#include <stan/math/opencl/kernel_generator.hpp>
+#include <stan/math/prim/err.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return the specified number of elements as a vector or row vector (same as
+ * input) from the back of the specified vector or row vector.
+ *
+ * @tparam T_x type of input kernel generator expression.
+ * @param x input kernel generator expression.
+ * @param n Size of return.
+ * @return The first n elements of v.
+ * @throw std::out_of_range if n is out of range.
+ */
+template <typename T_x,
+          typename = require_nonscalar_prim_or_rev_kernel_expression_t<T_x>>
```

Comment:
```suggestion
          require_nonscalar_prim_or_rev_kernel_expression_t<T_x>* = nullptr>
```
Trying to make require usage consistent across all stan math.

Reply:
Agree.